libmate-panel-applet/mate-panel-applet.c

```diff
@@ -2423,55 +2423,6 @@ static void mate_panel_applet_factory_main_finalized(gpointer data, GObject* obj
 	}
 }
 
-static int (*_x_error_func) (Display *, XErrorEvent *);
-
-static int
-_x_error_handler (Display *display, XErrorEvent *error)
-{
-	if (!error->error_code)
-		return 0;
-
-	/* If we got a BadDrawable or a BadWindow, we ignore it for now.
-	 * FIXME: We need to somehow distinguish real errors from
-	 * X-server-induced errors. Keeping a list of windows for which we
-	 * will ignore BadDrawables would be a good idea. */
-	if (error->error_code == BadDrawable ||
-	    error->error_code == BadWindow)
-		return 0;
-
-	return _x_error_func (display, error);
-}
-
-/*
- * To do graphical embedding in the X window system, MATE Panel
- * uses the classic foreign-window-reparenting trick. The
- * GtkPlug/GtkSocket widgets are used for this purpose. However,
- * serious robustness problems arise if the GtkSocket end of the
- * connection unexpectedly dies. The X server sends out DestroyNotify
- * events for the descendants of the GtkPlug (i.e., your embedded
- * component's windows) in effectively random order. Furthermore, if
- * you happened to be drawing on any of those windows when the
- * GtkSocket was destroyed (a common state of affairs), an X error
- * will kill your application.
- *
- * To solve this latter problem, MATE Panel sets up its own X error
- * handler which ignores certain X errors that might have been
- * caused by such a scenario. Other X errors get passed to gdk_x_error
- * normally.
- */
```

Comment:
I wonder if it's okay to remove all this code, given that comment... Won't X server throw some errors at us like it's described there? On the other hand, this code is 5 years old, and I don't know if it's still relevant for modern X versions.